src/machines/java/com/enderio/machines/common/blockentity/FluidTankBlockEntity.java

```diff
@@ -162,30 +260,37 @@ public FluidTank getFluidTank() {
     @Override
     public MachineInventoryLayout getInventoryLayout() {
         return MachineInventoryLayout
-            .builder(false)
-            .inputSlot((slot, stack) ->
-                (stack.getItem() instanceof BucketItem bucketItem && bucketItem.getFluid() != Fluids.EMPTY && !(bucketItem instanceof MobBucketItem)) || (
-                    !(stack.getItem() instanceof BucketItem) && stack.getCapability(CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY).isPresent()))
-            .outputSlot()
-            .inputSlot((slot, stack) -> stack.getItem() == Items.BUCKET || (!(stack.getItem() instanceof BucketItem) && stack
-                .getCapability(CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY)
-                .isPresent()))
-            .outputSlot()
-            .build();
+                .builder(false)
+                .inputSlot((slot, stack) -> (stack.getItem() instanceof BucketItem bucketItem
+                        && bucketItem.getFluid() != Fluids.EMPTY && !(bucketItem instanceof MobBucketItem))
+                        || (!(stack.getItem() instanceof BucketItem) && stack
+                                .getCapability(CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY).isPresent()))
+                .outputSlot()
+                .inputSlot((slot,
+                        stack) -> stack.getItem() == Items.BUCKET || (!(stack.getItem() instanceof BucketItem) && stack
+                                .getCapability(CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY)
+                                .isPresent()))
+                .outputSlot()
+                .build();
     }
 
     // region Serialization
 
     @Override
     public void saveAdditional(CompoundTag pTag) {
         super.saveAdditional(pTag);
-        pTag.put("fluid", fluidTank.writeToNBT(new CompoundTag()));
+        if (fluidTank.isEmpty()) {
+            pTag.remove(FluidHandlerBlockItemStack.FLUID_NBT_KEY);
```

Comment:
is it necessary to try calling remove? As far as my understanding goes, whenever save is called its a fresh tag, so existing keys shouldn't be present?